src/librustc_mir/borrow_check/mod.rs

```diff
@@ -849,7 +849,17 @@ impl<'cx, 'gcx, 'tcx> MirBorrowckCtxt<'cx, 'gcx, 'tcx> {
                         }
                         WriteKind::Move => {
                             error_reported = true;
-                            this.report_move_out_while_borrowed(context, place_span, &borrow)
+
+                            // If the move occurs due to a closure, then we want to show the span
+                            // of the place where it's used in the closure.
+                            let closure_span = this.find_closure_span(place_span.1, context.loc);
+                            if let Some((_, var_span)) = closure_span {
```

Comment:
The first span is the `args_span`, you could use it to add a secondary span label pointing at move || saying "moved into this closure". That way there's no ambiguity (think of the case of nested closures).